### PR TITLE
Ignore empty lines and exceptions when parsing CSV

### PR DESCRIPTION
**Pull request by falaki**
* Adds dependency on slf4j
* Ignores empty records form CSVParser
* Ignores records that CSVParser throws exception
* Bumps the version to 0.1.1

### CODE REVIEW

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 1]
```diff
@@ -35,6 +36,8 @@ case class CsvRelation protected[spark] (
     quote: Char,
     userSchema: StructType = null)(@transient val sqlContext: SQLContext) extends TableScan {
 
+  private val logger = LoggerFactory.getLogger(classOf[CsvRelation])
```
Comment (rxin) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: "classOf[CsvRelation]" -> "getClass"

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 2]
```diff
@@ -120,12 +120,24 @@ case class CsvRelation protected[spark] (
...
-      for (index <- 0 until schemaFields.length) {
-        row(index) = tokens.get(index)
+    iter.flatMap { line =>
+      try {
+        val records = CSVParser.parse(line, csvFormat).getRecords
+        if (records.isEmpty) {
+          logger.warn(s"Ignoring empty line: $line")
+          None
+        } else {
+          val tokens = records.head
```
Comment (aarondav) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: Right now we expect that each element of the RDD is a line, but if we just iterated over the records and returned them, would that make everyone happy? Could also avoid the initial if check.

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 3]
```diff
@@ -120,12 +120,26 @@ case class CsvRelation protected[spark] (
...
+          var index = 0
+          while (index < schemaFields.length) {
+            row(index) = tokens.get(index)
+            index = index + 1
+          }
+          Some(projection(row))
+        }
+      } catch {
+        case e: Exception =>
```
Comment (rxin) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: you probably want to match NonFatal http://www.scala-lang.org/files/archive/nightly/docs/library/index.html#scala.util.control.NonFatal$

**Comment by rxin** [comment]
this lgtm - in the next iteration if you want to improve performance, you probably want to remove the flatMap with option and just define an iterator returned by mapPartitions

**Comment by aarondav** [comment]
Iterator#flatMap is implemented efficiently, is it not?

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 4]
```diff
@@ -120,12 +121,26 @@ case class CsvRelation protected[spark] (
       schemaFields: Seq[StructField],
       projection: MutableProjection,
       row: GenericMutableRow): Iterator[Row] = {
-    iter.map { line =>
-      val tokens = CSVParser.parse(line, csvFormat).getRecords.head
-      for (index <- 0 until schemaFields.length) {
-        row(index) = tokens.get(index)
+    iter.flatMap { line =>
+      try {
+        val records = CSVParser.parse(line, csvFormat).getRecords
```
Comment (aarondav) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: to be specific, I was referring to this:

```scala
CSVParser.parse(line, csvFormat).getRecords.map { tokens =>
  var index = 0
  while (index < schemaFields.length) {
    row(index) = tokens.get(index)
    index = index + 1
  }
  (projection(row))
}
```

As this is already merged, though, we can do this some time later.

**Pull request merged**